woocommerce/woocommerce
# Improve hpos cli verbose output

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

~~Note~~: This PR is a followup to #38690 which will need to be merged first.

`print_r` is used when logging warnings/errors during the HPOS CLI migrate command:

```
Warning: Attempting to remigrate...
Warning: 1 error found: Array
(
    [3304] => Array
        (
            [0] => Array
                (
                    [column] => _order_total
                    [original_value] => 14.29
                    [new_value] => 1429
                )

        )

)
 when re-migrating order. Please review the error above.
```

Formatting this output as JSON will make it easier to read when there are a significant number of entries output:

```
Warning: Attempting to remigrate...
Warning: 1 error found: {
    "3304": [
        {
            "column": "_order_total",
            "original_value": "14.29",
            "new_value": "1429"
        }
    ]
} when re-migrating order. Please review the error above.
```

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. With posts authoritative, enable sync.
2. Turn off the sync when both HPOS and posts tables are in sync.
3. Edit any existing order, or add a new order so that the out of sync condition can be simulated.
4. Run the CLI command like so: `wp wc cot verify_cot_data --verbose --re-migrate`. In the output, you will first see the order that you have edited shown as error and out of sync. This will be followed by message: `Warning: Attempting to remigrate` and then order will be re-migrated.
5. Output will be JSON instead of print_r.
